src/interfaces/controllers/insert-trailer-registration.ts

```diff
@@ -0,0 +1,75 @@
+import { Request, Response, NextFunction } from 'express';
+import * as domain from '../../domain';
+import { DataAccess } from '../../utils/data-access';
+import { IRequestHandler } from './i-request-handler';
+import { insertValidator } from '../validators/insert-trailer-validator';
+
+export class InsertTrailerRegistration implements IRequestHandler {
+  private dao: DataAccess;
+
+  constructor(_dao: DataAccess) {
+    this.dao = _dao;
+  }
+
+  private validate(payload?: domain.TrailerRegistration): string {
+    const errors = insertValidator.validate(payload, { abortEarly: false }).error;
+    return errors?.message;
+  }
+
+  public async call(req: Request, res: Response, next: NextFunction) {
+    const trailerRegistration = req.body as domain.TrailerRegistration;
+    const errors = this.validate(trailerRegistration);
+    if (errors) {
+      next(new domain.HTTPError(400, errors));
+      return;
+    }
+
+    const { vin, make, certificateIssueDate } = trailerRegistration;
+    const vinOrChassisWithMake = vin.length === 17 ? vin : vin + make;
+
+    try {
+      const existingTrailerRegistration = await this.getTrailerRegistrationByVinOrChassisWithMake(vinOrChassisWithMake);
+      if (!existingTrailerRegistration) {
+        trailerRegistration.vinOrChassisWithMake = vinOrChassisWithMake;
+        const result = await this.dao.upsertTrailerRegistration(trailerRegistration);
+        res.status(200).send(result);
+        return;
+      }
+      const { archive, reasonForDeregistration, deregisterDate } = existingTrailerRegistration;
```

Comment:
I think it is best if we split this into a separate file that we load in there. I understand the requirement why we are doing this (AC3) however if the requirements get to change it will make this more maintainable if this was a separate use-case of the service for example.
We seem to have:
- deregistering a trailer
- registering a trailer (with archive)